apps/api-extractor/src/api/ExtractorConfig.ts

```diff
@@ -256,11 +296,104 @@ export class ExtractorConfig {
       throw new InternalError('Expected absolute path: ' + absolutePath);
     }
     if (Path.isUnderOrEqual(absolutePath, this.projectFolder)) {
-      return path.relative(this.projectFolder, absolutePath).replace(/\\/g, '/');
+      return Path.convertToSlashes(path.relative(this.projectFolder, absolutePath));
     }
     return absolutePath;
   }
 
+  /**
+   * Searches for the api-extractor.json config file associated with the specified starting folder,
+   * and loads the file if found.  This lookup supports
+   * {@link https://www.npmjs.com/package/@rushstack/rig-package | rig packages}.
+   *
+   * @remarks
+   * The search will first look for a package.json file in a parent folder of the starting folder;
+   * if found, that will be used as the base folder instead of the starting folder.  If the config
+   * file is not found in `<baseFolder>/api-extractor.json` or `<baseFolder>/config/api-extractor.json`,
+   * then `<baseFolder/config/rig.json` will be checked to see whether a
+   * {@link https://www.npmjs.com/package/@rushstack/rig-package | rig package} is referenced; if so then
+   * the rig's api-extractor.json file will be used instead.  If a config file is found, it will be loaded
+   * and returned with the `IExtractorConfigPrepareOptions` object. Otherwise, `undefined` is returned
+   * to indicate that API Extractor does not appear to be configured for the specified folder.
+   *
+   * @returns An options object that can be passed to {@link ExtractorConfig.prepare}, or `undefined`
+   * if not api-extractor.json file was found.
+   */
+  public static tryLoadForFolder(
+    options: IExtractorConfigLoadForFolderOptions
+  ): IExtractorConfigPrepareOptions | undefined {
+    const packageJsonLookup: PackageJsonLookup = options.packageJsonLookup || new PackageJsonLookup();
```

Comment:
It's weird for a library project to use `PackageJsonLookup.instance` because it has no idea how up-to-date the cache is.  Maybe the application or another library initialized it a very long time ago.  Compare the worst cases:

- The caller can opt-in to caching by providing `IExtractorConfigLoadForFolderOptions.packageJsonLookup`.  Worst case: The app runs slower.
- The caller can opt-out of caching which by default uses `PackageJsonLookup.instance`.  Worst case: The API mysteriously malfunctions, and there is no explicit setting anywhere communicating that a cache is in use